resources/default.rb

```diff
@@ -1,18 +1,170 @@
-attribute :properties_cookbook, kind_of: String, default: lazy { node['nexus3']['properties_cookbook'] }
-attribute :properties_source, kind_of: String, default: lazy { node['nexus3']['properties_source'] }
-attribute :properties_variables, kind_of: Hash, default: lazy { node['nexus3']['properties_variables'] }
-attribute :vmoptions_cookbook, kind_of: String, default: lazy { node['nexus3']['vmoptions_cookbook'] }
-attribute :vmoptions_source, kind_of: String, default: lazy { node['nexus3']['vmoptions_source'] }
-attribute :vmoptions_variables, kind_of: Hash, default: lazy { node['nexus3']['vmoptions_variables'] }
+property :instance_name, kind_of: String, name_attribute: true
+property :nexus3_user, kind_of: [String, NilClass], default: lazy { node['nexus3']['user'] }
+property :nexus3_group, kind_of: [String, NilClass], default: lazy { node['nexus3']['group'] }
+# property :nexus3_password, kind_of: [String, NilClass], sensitive: true # Admin password
+property :version, kind_of: String, default: lazy { node['nexus3']['version'] }
+property :url, kind_of: String, default: lazy { node['nexus3']['url'] }
+property :checksum, kind_of: String, default: lazy { node['nexus3']['checksum'] }
+property :nexus3_home, kind_of: String, default: lazy { node['nexus3']['home'] }
+property :path, kind_of: String, default: lazy { node['nexus3']['path'] }
+property :data, kind_of: String, default: lazy { node['nexus3']['data'] }
+property :properties_variables, kind_of: Hash, default: lazy { node['nexus3']['properties_variables'] }
+property :vmoptions_variables, kind_of: Hash, default: lazy { node['nexus3']['vmoptions_variables'] }
+
+load_current_value do
+  # TODO: check if user, group, etc. exist for each property or so. But: do we need to do so?
+end
+
+# TODO: use helpers to make this section smaller?
+action :install do
+  install_dir = ::File.join(path, "nexus-#{version}")
+
+  user new_resource.nexus3_user do
+    comment 'Nexus 3 user'
+    home nexus3_home
+    manage_home false # is linked to install_dir below
+    shell '/bin/bash'
+  end
+
+  group new_resource.nexus3_group do
+    members nexus3_user
+    append true
+  end
+
+  # Install Nexus3 software
+  include_recipe 'ark'
+
+  ark "nexus-#{new_resource.version}" do
+    action :put
+    url download_url
+    checksum new_resource.checksum
+    path new_resource.path
+    owner new_resource.nexus3_user
+    group new_resource.nexus3_group
+  end
+
+  # Setup directories
+  [install_dir, data, ::File.join(data, 'etc')].each do |dir|
+    directory dir do
+      recursive true
+      owner nexus3_user
+      group nexus3_group
+      mode '0755'
+    end
+  end
+
+  # Install configuration from templates
+  template ::File.join(install_dir, 'bin', 'nexus.rc') do
+    source 'nexus.rc.erb'
+    variables(user: nexus3_user)
+    mode '0644'
+    owner nexus3_user
+    group nexus3_group
+    cookbook 'nexus3'
+    notifies(:write, "log[#{instance_name} is restarting]", :delayed)
```

Comment:
Sounds good, although I don't want to restart the service until all files are in place, so :delayed seems the better way.